Realm/Realm/Handles/ObjectHandle.cs

```diff
@@ -67,7 +67,7 @@ private static class NativeMethods
             public static extern bool equals_object(ObjectHandle handle, ObjectHandle otherHandle, out NativeException ex);
 
             [DllImport(InteropConfig.DLL_NAME, EntryPoint = "object_get_obj_key", CallingConvention = CallingConvention.Cdecl)]
-            public static extern ObjKey get_obj_key(ObjectHandle handle, out NativeException ex);
+            public static extern Int64 get_obj_key(ObjectHandle handle, out NativeException ex);
```

Comment:
This is marginally related to this PR, but let's remove this API altogether. It's only used to calculate the hash code of `RealmObjectBase`. Instead, let's replace it with a `object_get_hashcode` and implement it natively. That way we can remove ObjKey from the .NET SDK entirely. Additionally, it gives us an opportunity to include the TableKey in the hashcode calculation.